src/core/etl/src/Flow/ETL/DSL/functions.php

```diff
@@ -795,6 +795,78 @@ function number_format(ScalarFunction $function, ?ScalarFunction $decimals = nul
     return new NumberFormat($function, $decimals, $decimalSeparator, $thousandsSeparator);
 }
 
+/**
+ * @psalm-suppress PossiblyInvalidIterator
+ *
+ * @param array<array<mixed>>|array<mixed|string> $data
+ */
+function array_to_row(array $data, EntryFactory $entryFactory = new NativeEntryFactory(), ?Schema $schema = null) : Row
```

Comment:
once this is implemented it can be used in array_to_rows I believe while iterating over rows to reduce the code duplication